File: web/src/app/modules/game-data/model/team.ts
export class Team {
  id: number;
  name: string;
  score: number;
  havePressed: boolean;
  timePressed: number;
  timeSynced: number;
  deviceId: number;
  quickest: boolean;
  winner: boolean;

  constructor(id: number, name: string, score: number) {
    this.id = id;
    this.name = name;
    this.score = score;
  }
}

File: core/src/main/java/eu/kotkas/villak/core/model/Team.java
package eu.kotkas.villak.core.model;

import eu.kotkas.villak.core.util.IdUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 * @author Kristen Kotkas
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Team implements Serializable {

  @Builder.Default
  private int id = IdUtil.getId();
  private String name;
  @Builder.Default
  private int score = 0;
  private boolean havePressed;
  @Builder.Default
  private Long timePressed = 0L;
  @Builder.Default
  private Long timeSynced = 0L;
  private Long deviceId;
  private boolean quickest;
  private boolean winner;

}
